### PR TITLE
`Development`: Update monaco-editor to 0.52.0

### PR DESCRIPTION
**Pull request by pzdr7**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).

#### Changes affecting Programming Exercises
- [x] **High priority**: I tested **all** changes and their related features with **all** corresponding user types on a test server configured with the **integrated lifecycle setup** (LocalVC and LocalCI).

### Motivation and Context

After upgrading to `0.52.0`, the Monaco editor would sometimes encounter errors related to its web worker, which would cause it to lose its language features.

Potentially related to https://github.com/microsoft/monaco-editor/issues/4694, although it's not 100% clear if this is the same error

### Description

- Adjusted the configuration to follow the `monaco-editor` esbuild sample (https://github.com/microsoft/monaco-editor/blob/main/samples/browser-esm-esbuild/build.js)
- Updated `monaco-editor` to version 0.52.0
- Fixed a small bug related to completers after a subtle change in how Monaco handles the `type` action

### Steps for Testing

- 1 Instructor
- 1 Course with:
  - 1 Programming exercise. The solution build must be finished and there must be at least one test case.
  - Communication enabled

1. Navigate to Course management > Your course > Exercises > Your programming exercise and click the name of the programming exercise.
2. Scroll down, click "Review changes" and verify that this does not load endlessly
3. Close the modal, scroll up, and click on "Edit"
4. Verify that you still get completions when typing in the problem statement editor.
5. Go to the communication view of your course and open a channel.
6. Verify that when you click on \@, \#, and "Exercise" in the Markdown editor action bar, the suggestion widget opens.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

Unchanged

### Screenshots

No UI changes

## Summary by CodeRabbit

- **New Features**
	- Introduced a new completion feature in the Monaco editor, enhancing user experience by suggesting completions based on context.
  
- **Bug Fixes**
	- Updated asset paths for the Monaco editor to improve resource loading during the build process.

- **Documentation**
	- Revised method documentation for clarity regarding text insertion and completion triggering.

- **Chores**
	- Updated the `monaco-editor` dependency to the latest version for improved performance and features.

### CODE REVIEW

File: src/main/webapp/app/shared/monaco-editor/model/actions/adapter/monaco-text-editor.adapter.ts [hunk 1]
```diff
@@ -152,8 +152,8 @@ export class MonacoTextEditorAdapter implements TextEditor {
         return this.editor.getDomNode() ?? undefined;
     }
 
-    typeText(text: string) {
-        this.editor.trigger('MonacoTextEditorAdapter::typeText', 'type', { text });
```
Comment (pzdr7) [inline src/main/webapp/app/shared/monaco-editor/model/actions/adapter/monaco-text-editor.adapter.ts:156]: Typing is no longer sufficient to trigger completions, so I added a method that explicitly brings up the suggest widget

File: src/main/webapp/app/core/config/monaco.config.ts [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
-            /*
-             * This is the AMD-based service worker, which comes bundled with a few special workers for selected languages.
-             * (e.g.: javascript, typescript, html, css)
-             *
-             * It is also possible to use an ESM-based approach, which requires a little more setup and case distinctions in this method.
-             * At the moment, it seems that the ESM-based approaches are incompatible with the Artemis client, as they would require custom builders.
-             * Support for custom builders was removed in #6546.
-             */
-            return 'vs/base/worker/workerMain.js';
+        getWorkerUrl: function (moduleId, label) {
```
[the remark this answers is not in the document]
Comment (pzdr7) [reply src/main/webapp/app/core/config/monaco.config.ts:7]: this is an mjs file, not a ts file

[the remark this answers is not in the document]
Comment (pzdr7) [reply src/main/webapp/app/core/config/monaco.config.ts:7]: Sorry, remove that learning

[the remark this answers is not in the document]
Comment (pzdr7) [reply src/main/webapp/app/core/config/monaco.config.ts]: I added the suggested change

**Review by JohannesWt — APPROVED** [review]
Code

**Review by magaupp — APPROVED** [review]
Tested on TS4.
"Review changes" diff view loads properly. Test cases are completed in the problem statement editor. Clicking the @, #, and "Exercise" buttons opens the completion popup.

Unrelated to this PR:
I noticed that the @ name completion does not show the correct names in a PostgreSQL deployment.
Also, clicking on "Disable split view" in the "Review Changes" modal does not work at the first try.

**Review by az108 — APPROVED** [review]
Tested on TS4 on a programming exercise, review changes was possible and problem statement editor was working as expected and markdown editor in communications section is also working 👍

**Pull request merged**